src/route.c

```diff
@@ -277,22 +246,42 @@ static int route_add_lcore(struct in_addr* dest,uint8_t netmask, uint32_t flag,
     return EDPVS_INVAL;
 }
 
+/* del route node in list, then mbuf next will never find it;
+ * route4_put will delete route when refcnt is 0.
+ * refcnt:
+ * 1, new route is set to 0;
+ * 2, add list will be 1;
+ * 3, find route and ref it will +1;
+ * 4, put route will -1;
+ */
 static int route_del_lcore(struct in_addr* dest,uint8_t netmask, uint32_t flag,
               struct in_addr* gw, struct netif_port *port,
               struct in_addr* src, unsigned long mtu,short metric)
 {
     struct route_entry *route = NULL;
-    int error;
+
     if(flag & RTF_LOCALIN || (flag & RTF_KNI)){
         route = route_local_lookup(dest->s_addr, port);
-        error = route_local_del(route);
-        return error;
+        if (!route)
+            return EDPVS_NOTEXIST;
+        list_del(&route->list);
+        rte_atomic32_dec(&route->refcnt);
+        rte_atomic32_dec(&this_num_routes);
+        route4_put(route);
```

Comment:
Is it better to decrease `this_num_routes`  when route is freed?

Reply:
this_num_routes just read routes in list. When route is delete from list, this_num_routes should be decreased, or this_num_routes will not be correct.